safemysql.class.php

```diff
@@ -170,6 +174,39 @@ public function query()
 		return $this->rawQuery($this->prepareQuery(func_get_args()));
 	}
 
+
+	/**
+	 * Run the provided callback inside a transaction
+	 * @param  callable $callback
+	 * @param  [boolean] $can_retry If the whole transaction should be retried, if it fails due to a deadlock
+	 * @return mixed whatever the callback returns
+	 */
+	public function transaction($callback, $can_retry = false)
+	{
+		$this->query('START TRANSACTION');
+
+		// If there is a deadlock it may be permissable to rerun the entire callback, but individual statements
+		// within it should not be rerun
+		$retry_on_deadlock_status = $this->retryOnDeadlock;
+		$this->retryOnDeadlock = false;
+
+		try
+		{
+			$result = $this->retryIfDeadlocked($callback, $can_retry);
+			$this->query('COMMIT');
+			return $result;
+		} catch (Throwable $e)
```

Comment:
- [ ] Hmm. This will only work in PHP 7, and the library still claims to only need >= PHP 5.x in its composer.json. We should either increase the minimum version to PHP 7, or use [this technique](http://php.net/manual/en/language.errors.php7.php#119652) for supporting both languages.